test/unit-test/mqtt_agent_utest.c

```diff
@@ -258,6 +269,45 @@ MQTTStatus_t MQTT_ProcessLoop_stub( MQTTContext_t * pContext,
     return MQTTSuccess;
 }
 
+/**
+ * @brief A stub for MQTT_ProcessLoop function to be used to test multiple calls to MQTT_ProcessLoop.
+ */
+MQTTStatus_t MQTT_ProcessLoop_stub_multiple( MQTTContext_t * pContext,
+                                             uint32_t timeoutMs )
+{
+    MQTTPacketInfo_t packetInfo;
+    MQTTDeserializedInfo_t deserializedInfo;
+    MQTTAgentContext_t * pMqttAgentContext;
+    MQTTStatus_t status;
+
+    packetInfo.type = packetType;
+    deserializedInfo.packetIdentifier = packetIdentifier;
+
+    pContext->appCallback( pContext, &packetInfo, &deserializedInfo );
+    pMqttAgentContext = ( MQTTAgentContext_t * ) pContext;
+    pMqttAgentContext->packetReceivedInLoop = false;
+
+    if( processLoopCounter == 0 )
+    {
+        pMqttAgentContext->packetReceivedInLoop = true;
+        status = MQTTSuccess;
+    }
+    else if( processLoopCounter == 1 )
+    {
+        pMqttAgentContext->packetReceivedInLoop = true;
```

Comment:
You don't need to set this flag since it's set when you call `pContext->appCallback()`. It would make more sense to just not call the callback when you want to set it to false.